src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,990 @@
+            if (2 === $pos && ctype_alpha($path[0]) && ':' === $path[1]) {
+                return $scheme.substr($path, 0, 3);
+            }
+
+            return $scheme.substr($path, 0, $pos);
+        }
+
+        return '';
+    }
+
+    /**
+     * Returns canonical path of the user's home directory.
+     *
+     * Supported operating systems:
+     *
+     *  - UNIX
+     *  - Windows8 and upper
+     *
+     * If your operation system or environment isn't supported, an exception is thrown.
+     *
+     * The result is a canonical path.
+     *
+     * @return string The canonical home directory
+     *
+     * @throws RuntimeException If your operation system or environment isn't supported
+     *
+     * @since 2.1 Added method.
+     */
+    public static function getHomeDirectory(): string
+    {
+        // For UNIX support
+        if (getenv('HOME')) {
+            return static::canonicalize(getenv('HOME'));
+        }
+
+        // For >= Windows8 support
+        if (getenv('HOMEDRIVE') && getenv('HOMEPATH')) {
+            return static::canonicalize(getenv('HOMEDRIVE').getenv('HOMEPATH'));
+        }
+
+        throw new RuntimeException("Your environment or operation system isn't supported");
+    }
+
+    /**
+     * Returns the root directory of a path.
+     *
+     * The result is a canonical path.
+     *
+     * @param string $path A path string.
+     *
+     * @return string The canonical root directory. Returns an empty string if
+     *                the given path is relative or empty.
+     *
+     * @since 1.0 Added method.
+     * @since 2.0 Method now fails if $path is not a string.
+     */
+    public static function getRoot(string $path): string
+    {
+        if ('' === $path) {
+            return '';
+        }
+
+        // Maintain scheme
+        if (false !== ($pos = strpos($path, '://'))) {
+            $scheme = substr($path, 0, $pos + 3);
+            $path = substr($path, $pos + 3);
+        } else {
+            $scheme = '';
+        }
+
+        // UNIX root "/" or "\" (Windows style)
+        if ('/' === $path[0] || '\\' === $path[0]) {
+            return $scheme.'/';
+        }
+
+        $length = strlen($path);
+
+        // Windows root
+        if ($length > 1 && ctype_alpha($path[0]) && ':' === $path[1]) {
+            // Special case: "C:"
+            if (2 === $length) {
+                return $scheme.$path.'/';
+            }
+
+            // Normal case: "C:/ or "C:\"
+            if ('/' === $path[2] || '\\' === $path[2]) {
+                return $scheme.$path[0].$path[1].'/';
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     * Returns the file name from a file path.
+     *
+     * @param string $path The path string.
+     *
+     * @return string The file name.
+     *
+     * @since 1.1 Added method.
+     * @since 2.0 Method now fails if $path is not a string.
+     */
+    public static function getFilename(string $path): string
```

Comment:
do we need this :) https://3v4l.org/4mphY

Reply:
did you try this on Windows? I'm pretty sure there was a reason for this.

Reply:
It was added here: https://github.com/webmozart/path-util/pull/3